apps/client/vite.config.ts

```diff
@@ -3,28 +3,37 @@ import tailwindcss from "@tailwindcss/vite";
 import router from "@tanstack/router-plugin/vite";
 import basicSSL from "@vitejs/plugin-basic-ssl";
 import react from "@vitejs/plugin-react";
-import { defineConfig } from "vite";
+import { defineConfig, loadEnv } from "vite";
 import tsConfigPaths from "vite-tsconfig-paths";
 
-export default defineConfig({
-	server: {
-		port: 4483,
-		host: "0.0.0.0",
-		https: {},
-	},
-	plugins: [
-		tsConfigPaths(),
-		router({
-			target: "react",
-			autoCodeSplitting: true,
-		}),
-		react(),
-		tailwindcss(),
-		basicSSL(),
-	],
-	resolve: {
-		alias: {
-			"@": path.resolve(__dirname, "./src"),
+export default defineConfig(({ mode }) => {
+	const env = loadEnv(mode, process.cwd(), "");
+	return {
+		server: {
+			port: 4483,
+			host: "0.0.0.0",
+			https: {},
+			proxy: {
+				"/trpc": {
+					target: env.VITE_PUBLIC_API_URL,
```

Comment:
It might be best to create a `VITE_PRIVATE_API_URL` for the proxy to use that points to the internal API URL. This variable would only ever be used in development environments since we won't use the Vite proxy in prod. Also, it's important to remember that in prod we might not be using `/trpc` as the endpoint for the TRPC API, it can possibly be at a different location or sub-domain.

```suggestion
					target: env.VITE_PRIVATE_API_URL,
```

Reply:
Addressed: The url used for the proxy's target is now `VITE_PRIVATE_API_URL`.